### ISSUE TITLE
Refactor agent creation into creation events

### ISSUE THREAD

**Issue by Ozaq** [JuPedSim/jpscore#876]
Agents should no longer be added to the simulation from the simulation itself. Agents should be added as a result from a `CreateAgentEvent`.

For this we need: 
* A revamped EventManager that works on a std::variant of event types. 
* Be able to retrieve events by timestamp `t`, e.g. get all non-consumed events until `t`
* Visitor to connect events to actions (e.g. https://en.cppreference.com/w/cpp/utility/variant/visit)

Conceptually this should allow to be used like this:

```cpp
while(cond) {
  auto events = evtmgr.NextEvents(time);
  for(const auto& evt : events) {
    std::visit(...evt...);
  }
  sim.Iterate();
}
```

As part of this we also remove pedestrian distribution from Simulation and seed all events before starting the simulation.

Non-goal: This is not supposed to be able to handle millions of events, all should be kept in memory.

**Comment by Ozaq** [comment]
Done.
